src/request/selectors/RequestDetailRequestSelectors.ts

```diff
@@ -10,31 +9,50 @@ const getMiddlewareState = (state: IRequestState) => state.detail.request.middle
 export const getRequest = (state: IRequestState) => state.detail.request;
 export const getUrl = (state: IRequestState) => state.detail.request.url;
 
+interface IMiddleware {
+    name: string;
+    packageName: string;
+    headers: { [key: string]: { value: string, isCurrent: boolean } };
+}
+
 interface IFlattenedMiddleware {
-    depth: number,
-    middleware: { name: string, packageName: string, headers: { [key: string]: string } }
+    depth: number;
+    middleware: IMiddleware;
 }
 
-function flattenMiddlewareRecursive(middleware: IRequestDetailRequestMiddlewareState[], middlewareArray: IFlattenedMiddleware[], depth: number): void {
+function flattenMiddlewareRecursive(middleware: IRequestDetailRequestMiddlewareState[], middlewareArray: IFlattenedMiddleware[], currentHeaders: { [key: string]: IMiddleware }, depth: number): void {
 
     middleware.forEach(middlewareItem => {
+        const m = {
+            name: middlewareItem.name,
+            packageName: middlewareItem.packageName,
+            headers: _.mapValues(middlewareItem.headers, value => { return { value: value.value, isCurrent: false }; })
+        };
+
+        _.forEach(middlewareItem.headers, (value, key) => {
+            currentHeaders[key] = value.wasSet ? m : undefined;
+        });
+
         middlewareArray.push({
             depth: depth,
-            middleware: {
-                name: middlewareItem.name,
-                packageName: middlewareItem.packageName,
-                headers: middlewareItem.headers
-            }
+            middleware: m
```

Comment:
Was originally `middleware` but the linter complained about it being a "shadow variable" so shortened it as I couldn't think of something better at the time.  I've updated it to `newMiddleware`.